docs/connector-development/config-based/declarative-oauth.md

```diff
@@ -0,0 +1,730 @@
+       complete_oauth_output_specification:
+         required:
+           - access_token
+
+```
+
+Example URL:
+`https://yourconnectorservice.com/oauth/consent?client_id=YOUR_CLIENT_ID_123&redirect_uri=https://cloud.airbyte.com&state=some_random_state_string&scope=my_scope_A:read,my_scope_B:read`
+
+#### Example using an url-encoded / url-decoded `scope` parameter
+You can make the `scope` paramter `url-encoded` by specifying the `pipe` (|) + `urlEncoder` or `urlDecoder` in the target url.
+It would be pre-formatted and resolved into the `url-encoded` string before being replaced for the final resolved URL. 
+
+```diff
+--- a/Users/baz/gl/airbyte/docs/connector-development/config-based/understanding-the-yaml-file/declarative_oauth_examples/scopes.yml.yml
++++ b/Users/baz/gl/airbyte/docs/connector-development/config-based/understanding-the-yaml-file/declarative_oauth_examples/scopes.yml
+@@ -80,10 +80,10 @@ spec:
+     oauth_config_specification:
+       oauth_connector_input_specification:
+         consent_url: >-
+-          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{
+-          redirect_uri_value }}&state={{ state_value }}&scope={{scope_value}}
++          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{ redirect_uri_value }}&state={{ state_value }}&scope={{ {{scope_value}} | urlEncoder }}
+         access_token_url: >-
+           https://yourconnectorservice.com/oauth/token?client_id={{client_id_value}}&client_secret={{client_secret_value}}&code={{auth_code_value}}
+-        scope: my_scope_A:read,my_scope_B:read
++        scope: my_scope_A:read my_scope_B:read
+       complete_oauth_output_specification:
+         required:
+           - access_token
+
+```
+
+Example URL:
+`https://yourconnectorservice.com/oauth/consent?client_id=YOUR_CLIENT_ID_123&redirect_uri=https://cloud.airbyte.com&state=some_random_state_string&scope=my_scope_A%3Aread%20my_scope_B%3Aread`
+
+
+### Advanced Case: get code challenge from the `state_value` and set as a part of the `consent_url` 
+Imagine that the OAuth flow is updated so that you need to generate the `code challenge`, using `SHA-256` hash and include this as a query parameter in the `consent_url` to get to the `Consent Screen`, before moving forward with authentication.
+
+#### Example Declarative OAuth Change
+```diff
+--- base_oauth.yml
++++ base_oauth.yml
+@@ -80,10 +80,10 @@ spec:
+     oauth_config_specification:
+       oauth_connector_input_specification:
+         consent_url: >-
+-          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{
+-          redirect_uri_value }}&state={{ state_value }}
++          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{
++          redirect_uri_value }}&state={{ state_value }}&code_challenge={{ {{state_value}} | codeChanllengeS256 }}
```

Comment:
💎 Thanks for this example
❓ is `codeChanllengeS256` a typo? and should be `codeChallengeS256`?
❓ is `codeChanllengeS256` essentially just an encrypt function? Would we be better off to name it `encryptSha256`?

Reply:
That was a typo, I corrected it.
I was following the namings from `base64Encoder` but didn't want to introduce the `codeChallengeEncoder` or something like that, so I made that precisely named what it does: `codeChallengeS256`. If this is not a blocker, I'm leaving it as is.